deployments/report-server-bucket-storage.nix

```diff
@@ -0,0 +1,60 @@
+{ IOHKaccessKeyId, ... }:
+
+{
+  # this file will create an S3 bucket, and IAM role with access to that bucket
+  # then give the report-server machine access to that role, and mount the bucket within the instance
+  report-server = { pkgs, resources, ... }: {
+    # nixops can only set this at creation, it will need to be manualy set
+    deployment.ec2.instanceProfile = resources.iamRoles.report-role.name;
+    fileSystems."/var/lib/report-server" = {
+      # mount unsets PATH when running this
+      device = "/run/current-system/sw/bin/s3fs#${resources.s3Buckets.report-bucket-3.name}";
+      fsType = "fuse";
+      options = [ "_netdev" "allow_other" "iam_role" ];
+    };
+    environment.systemPackages = [ pkgs.s3fs ];
+  };
+  resources = {
+    s3Buckets = {
+      # if moved to another region, you must rename the bucket
+      report-bucket-3 = { config, uuid, ... }: {
```

Comment:
Hmm, we should probably pick a better name :) Maybe `report-server-logs`?

Reply:
also keep in mind that nixops will mangle the name a bit, https://github.com/NixOS/nixops/blob/master/nix/s3-bucket.nix#L10